### ISSUE TITLE
non-admin user can execute iclienthints command

### ISSUE THREAD

**Issue by mstfdkmn** [irods/irods#7785]
I am not sure if this is an intended result but a non admin user is able to execute the `iclienthints` command.

**Comment by alanking** [comment]
Just some observations / commentary...

The following information is displayed in 4.3.2:
- hash_scheme configuration
- match_hash_policy configuration
- All installed plugins
- All rules defined in core.re (I'm using a default installation in the testing environment... this may also reveal other configured rulebases)
- Available specific_queries
- strict_acls configuration

I know regular users can use specific queries and see the specific queries in the catalog. The others may be beneficial for savvy users writing their own client applications or rules.

**Comment by mstfdkmn** [comment]
Correct, no secret in the output. However, I can't see any added value.  Would not letting regular users see all available rules (including custom ones) and msis in the server create a possibility that users might wrongly call an unnecessary function/msi/rule? 

However IMO it would be useful if regular users can see only marked rules/functions (for example admins can add some an identifier) in order to interact with those identified functions in the server.

**Comment by trel** [comment]
> However IMO it would be useful if regular users can see only marked rules/functions (for example admins can add some identifiers) in order to interact with those identified functions in the server.

We've talked about this before.  There is currently no distinction like this in the server.  All rules are equal at this time.

A rule could always enforce something like this in the rule_text itself - exit early if the calling user is not a rodsadmin or in a list of users or some other criteria.  If you'd like to open an issue for this as a separate discussion, along with a use case, please do.

**Comment by alanking** [comment]
The suggested feature (differentiating rules) feels out of scope for 4.3.5 if we decide to implement it at all.

**Comment by trel** [comment]
agreed.   i don't see us implementing this.   i vote to close.

**Comment by korydraughn** [comment]
Agreed. Close it.
